src/demo/apps/LauncherApp.tsx

```diff
@@ -16,14 +16,23 @@ export function LauncherApp(): React.ReactElement {
     React.useEffect(() => {
         fdc3.findIntent(null!)
             .then(async (appIntent) => setApplications(appIntent.apps))
-            .catch(console.log);
+            .catch(console.error);
     }, []);
 
-    const openApp = (app: Application) => {
+    const openApp = async (app: Application) => {
         console.log(`Opening app ${app.title}`);
-        fdc3.open(app.appId)
-            .then(() => console.log(`Opened app ${app.title}`))
-            .catch(console.log);
+        try {
+            await fdc3.open(app.appId);
+            console.log(`Opened app ${app.title}`);
+        } catch (e) {
+            // Stringifying an `Error` omits the message!
+            const error: any = {
+                message: e.message,
+                ...e
+            };
+            console.error(e, error);
+            alert(`openApp threw an error!\n${JSON.stringify(error)}`);
+        }
```

Comment:
an alert is pretty drastic. If we are going to do that, we need to be a lot clearer - it will need a lot of diagnostics. I think I'd rather keep it in the console.

Reply:
it was meant as a quick and dirty way to test the FDC3Errors in the UI :) - and for anyone testing without the console open. 

Happy to remove by now. Maybe we should add something to the backlog to add some nice error handling?

Reply:
Yes, that's a good plan.